client/src/cordova/apple/xcode/ios/Outline/Classes/AppKitBridge/AppKitController.swift

```diff
@@ -16,13 +16,64 @@ import AppKit
 import ServiceManagement
 
 class AppKitController: NSObject {
+    static let shared = AppKitController()
     private var statusItemController: StatusItemController?
+    private var windowCloseObservers: [NSWindow: NSObjectProtocol] = [:]
 
     override public required init() {
         super.init()
-
-        // Indicates that the application is an ordinary app that appears in the Dock and may have a user interface.
         NSApp.setActivationPolicy(.regular)
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(windowDidBecomeMain(_:)),
+            name: NSWindow.didBecomeMainNotification,
+            object: nil
+        )
+    }
+
+    @objc private func windowDidBecomeMain(_ notification: Notification) {
+        guard let window = notification.object as? NSWindow else {
+            return
+        }
+        setDockIconVisible(true)
+        // Register observer only once (prevent duplicate registration)
+        if windowCloseObservers[window] == nil {
+            observeWindowClose(for: window)
+        }
```

Comment:
This can be simplified as well.

```suggestion
        if notification.object is NSWindow {
            setDockIconVisible(true)
        }
```